src/object/srv_obj_migrate.c

```diff
@@ -1301,6 +1399,27 @@ migrate_cont_iter_cb(daos_handle_t ih, d_iov_t *key_iov,
 	}
 
 
+	/* Check if the container uuid is in the init hash table
+	 * If it is, this container has already had at least one object in it
+	 * migrated during this cycle, so it does not need to be created
+	 *
+	 * If this migration process is part of reintegration, the container
+	 * should first be deleted before being re-created
+	 *
+	 * TODO: Hook up reintegration boolean
+	 */
+	link = d_hash_rec_find(&tls->mpt_cont_init_tab, cont_uuid,
+			       sizeof(uuid_t));
```

Comment:
If we can have a barrier for all migration, i.e. local migration should start after local cleanup (deleting pool/containers). do we still need this hash table?

Reply:
We wouldn't - but we'd need some kind of way to know all of the containers that were going to have migrated objects ahead of time. I don't believe rebuild works this way today.

Reply:
oh, the object list should include container uuid, so it can just try to open/create the container, before migrate the object.
 I actually means the destroy part. so if we can set a barrier to delete all existing containers of the pool before migration, then we do not need this hash-table at all.

Reply:
oh, but we do not need resolve it in this patch.